src/module/actor/sheet/base.ts

```diff
@@ -633,24 +633,28 @@ abstract class ActorSheetPF2e<TActor extends ActorPF2e> extends ActorSheet<TActo
             (element): element is string => !!element
         );
 
-        const filter: Record<string, string[]> = {};
+        const eqTab = game.pf2e.compendiumBrowser.tabs.equipment;
+        const filter = await eqTab.getFilterData();
+        const { checkboxes } = filter;
+
         for (const filterCode of checkboxesFilterCodes) {
             const splitValues = filterCode.split("-");
             if (splitValues.length !== 2) {
-                console.error(
-                    `Invalid filter value for opening the compendium browser:\n'${JSON.stringify(
-                        checkboxesFilterCodes
-                    )}'`
-                );
+                console.error(`Invalid filter value for opening the compendium browser: "${filterCode}"`);
                 return;
```

Comment:
Can this be changed to an `ErrorPF2e` throw?

Reply:
Done. I've also improved the sheet listeners a bit more.